opentracing-api-extensions-tracer-autoconfigure/src/main/java/io/opentracing/contrib/api/tracer/autoconfigure/TracerBeanPostProcessor.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.api.tracer.autoconfigure;
+
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Configuration;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.api.TracerObserver;
+import io.opentracing.contrib.api.tracer.APIExtensionsTracer;
+
+@Configuration
+public class TracerBeanPostProcessor implements BeanPostProcessor {
+
+    private static final Log log = LogFactory.getLog(TracerBeanPostProcessor.class);
+
+    @Autowired(required=false)
+    private Set<TracerObserver> tracerObservers;
+
+    @Override
+    public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+        if (tracerObservers != null && bean instanceof Tracer) {
+            boolean observerFound = false;
+            APIExtensionsTracer tracer = new APIExtensionsTracer((Tracer)bean);
+            for (TracerObserver observer : tracerObservers) {
+                if (observer != null) {
+                    observerFound = true;
+                    tracer.addTracerObserver(observer);
+                }
+            }
+            if (observerFound) {
+                log.info("Use extensions API tracer with observers=" + tracerObservers);
+                return tracer;
```

Comment:
@objectiser I think @pavolloffay has a point in case there's a pre-registered `GlobalTracer`.
I agree with him that that should always take precedence in this case.
I added PR for review: https://github.com/opentracing-contrib/java-tracerresolver/pull/16

As it happens I also agree _with you_ that registering the `GlobalTracer` should _happen after_ the resolving (and potential wrapping).